src/lib.rs

```diff
@@ -23,6 +23,8 @@
 //! software.
 //!
 
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
as suggested also by @Kixunil there is no need to prevent use of std in tests

`#![cfg_attr(and(not(feature = "std"), not(test)), no_std)]`